src/cysignals/signals.pyx

```diff
@@ -108,16 +106,29 @@ cdef public int sig_raise_exception "sig_raise_exception"(int sig, const char* m
         if msg is NULL:
             msg = "Floating point exception"
         PyErr_SetString(FloatingPointError, msg)
-    elif sig == SIGBUS:
-        if msg is NULL:
-            msg = "Bus error"
-        PyErr_SetString(SignalError, msg)
     elif sig == SIGSEGV:
         if msg is NULL:
             msg = "Segmentation fault"
         PyErr_SetString(SignalError, msg)
-    else:
-        raise SystemError(f"unknown signal number {sig}")
+    IF UNAME_SYSNAME != 'Windows':
+        if sig == SIGHUP:
+            # Redirect stdin from /dev/null to close interactive sessions
+            _ = freopen("/dev/null", "r", stdin)
+            # This causes Python to exit
+            raise SystemExit
+        elif sig == SIGALRM:
+            raise AlarmInterrupt
+        elif sig == SIGBUS:
+            if msg is NULL:
+                msg = "Bus error"
+            PyErr_SetString(SignalError, msg)
+    ELSE:
+        if sig == 128:  # Used for signals that map to FPE
+            msg = "Pari Error"
```

Comment:
`Pari Error`? Where does that come from?

Reply:
All the "pari" stuff definitely needs to be taken for a first pass.  If there are places where pari needs to be able to hook in, that should be done in a more generic manner.